packages/app/src/pages/dashboard/migrate.tsx

```diff
@@ -0,0 +1,314 @@
+import { StakingCard } from 'sections/dashboard/Stake/card'
+import EscrowTable from 'sections/dashboard/Stake/EscrowTable'
+import StakingPortfolio, { StakeTab } from 'sections/dashboard/Stake/StakingPortfolio'
+import { useAppDispatch, useAppSelector } from 'state/hooks'
+import {
+	approveKwentaToken,
+	fetchClaimableRewards,
+	fetchEscrowData,
+	fetchEscrowV2Data,
+	fetchStakingData,
+	fetchStakingV2Data,
+	getReward,
+	stakeKwentaV2,
+	unstakeKwenta,
+} from 'state/staking/actions'
+import {
+	selectClaimableBalance,
+	selectClaimableBalanceV2,
+	selectIsKwentaTokenApprovedV2,
+	selectKwentaBalance,
+	selectKwentaRewards,
+	selectStakedEscrowedKwentaBalance,
+	selectStakedEscrowedKwentaBalanceV2,
+	selectStakedKwentaBalance,
+	selectStakedKwentaBalanceV2,
+	selectTotalVestable,
+	selectTotalVestableV2,
+} from 'state/staking/selectors'
+import { selectWallet } from 'state/wallet/selectors'
+import media from 'styles/media'
+
+import { StakingCards } from './staking'
+
+type MigrateComponent = React.FC & { getLayout: (page: ReactNode) => JSX.Element }
+
+const MigratePage: MigrateComponent = () => {
+	const { t } = useTranslation()
+	const router = useRouter()
+	const dispatch = useAppDispatch()
+	const walletAddress = useAppSelector(selectWallet)
+	const kwentaBalance = useAppSelector(selectKwentaBalance)
+	const claimableBalance = useAppSelector(selectClaimableBalance)
+	const claimableBalanceV2 = useAppSelector(selectClaimableBalanceV2)
+	const stakedKwentaBalance = useAppSelector(selectStakedKwentaBalance)
+	const stakedKwentaBalanceV2 = useAppSelector(selectStakedKwentaBalanceV2)
+	const stakedEscrowedKwentaBalance = useAppSelector(selectStakedEscrowedKwentaBalance)
+	const stakedEscrowedKwentaBalanceV2 = useAppSelector(selectStakedEscrowedKwentaBalanceV2)
+	const totalVestable = useAppSelector(selectTotalVestable)
+	const totalVestableV2 = useAppSelector(selectTotalVestableV2)
+	const kwentaStakingV2Approved = useAppSelector(selectIsKwentaTokenApprovedV2)
+	const kwentaRewards = useAppSelector(selectKwentaRewards)
+
+	const handleGetReward = useCallback(() => {
+		dispatch(getReward())
+	}, [dispatch])
+
+	const tabQuery = useMemo(() => {
+		if (router.query.tab) {
+			const tab = router.query.tab as StakeTab
+			if (Object.values(StakeTab).includes(tab)) {
+				return tab
+			}
+		}
+		return null
+	}, [router])
+
+	const handleUnstakeKwenta = useCallback(
+		() => dispatch(unstakeKwenta(wei(stakedKwentaBalance).toBN())),
+		[dispatch, stakedKwentaBalance]
+	)
+
+	const handleStakeKwentaV2 = useCallback(() => {
+		if (!kwentaStakingV2Approved) {
+			dispatch(approveKwentaToken('kwentaStakingV2'))
+		} else {
+			dispatch(stakeKwentaV2(wei(kwentaBalance).toBN()))
+		}
+	}, [dispatch, kwentaBalance, kwentaStakingV2Approved])
+
+	const [, setCurrentTab] = useState(tabQuery ?? StakeTab.Staking)
+
+	useEffect(() => {
+		if (!!walletAddress) {
+			dispatch(fetchStakingData()).then(() => {
```

Comment:
I think a better approach would be to create an action `fetchStakeMigrateData` and have it perform these fetches under there. This way we can use `useFetchAction` hook here with walletAddress as a dep and we'll also be able to track state of the action and any errors by using `builder.addCase(fetchStakeMigrateData.rejected)` etc in the reducer.

Reply:
Good idea. Updated.